posts/views.py

```diff
@@ -214,6 +216,19 @@ def unlike_post_view(request: HttpRequest, pk: int):
     return redirect(Post.objects.get(pk=pk).get_absolute_url())
 
 
+def share_post_view(request: HttpRequest, pk: int):
+
+    # duplicate the post
+    shared_post = Post.objects.get(pk=pk)
+    shared_post.pk = None
+    shared_post.save()
+    author = User.objects.get(username=request.user)
+    shared_post.author = author
+    shared_post.source = request_uri
+    shared_post.save()
```

Comment:
```suggestion
    shared_post = Post.objects.get(pk=pk)
    shared_post.pk = None
    shared_post.save()
    author = User.objects.get(username=request.user)
    shared_post.author = author
    shared_post.source = request_uri
    shared_post.save()
```
Please update this to Post.objects.create similar to how we do them in tests.

To add the source field, we might want to add a new field that's a foreign key to another post [reference](https://docs.djangoproject.com/en/dev/ref/models/fields/#foreignkey)

Reply:
it's almost done. I'm working on tests right now. Was wondering if there should also be a model for shared posts, so that we can discern original vs shared posts. Or else it looks like this: 
![image](https://user-images.githubusercontent.com/34114501/161485982-b1eb8a4e-ab05-44ae-91cf-9bd3a67b31de.png)

Reply:
or adding a new field called shared_author is sth that I am thinking of as well.

Reply:
Then it might not be a bad idea to just create a SharedPost model as follows
```py
author: Author
post: Post
```
Then we'll just render this differently, wdyt

Reply:
This way we don't have to duplicate the Post object

Reply:
yea, sounds alright